agent/util-scripts/pbench-register-tool-set

```diff
@@ -1,119 +1,153 @@
-if [ $? -ne 0 ]; then
-	printf "\n"
-	printf "%s\n" $*
-	printf "$script_name: you specified an invalid option or forgot a required option value\n\n"
-	usage
+opts=$(getopt -q -o ht:g:i:r:l: --longoptions "help,toolset:,group:,interval:,no-install,remote:,remotes:,label:,labels:" -n "getopt.sh" -- "${@}")
+if [[ ${?} -ne 0 ]]; then
+	printf -- "\n${*}\n" >&2
+	printf -- "${script_name}: you specified an invalid option\n\n" >&2
+	usage >&2
 	exit 1
 fi
-eval set -- "$opts";
+
+eval set -- "${opts}"
 while true; do
-	case "$1" in
-		--toolset)
-		shift;
-		if [ -n "$1" ]; then
-			toolset="$1"
-			shift;
+	case "${1}" in
+	-t|--toolset)
+		shift
+		if [[ -n "${1}" ]]; then
+			toolset="${1}"
+			shift
 		fi
 		;;
-		-r|--remote)
-		shift;
-		if [ -n "$1" ]; then
-			remote="--remote=$1"
-			shift;
+	-r|--remote|--remotes)
+		shift
+		if [[ -n "${1}" ]]; then
+			remotes_arg=" --remotes=${1}"
+			shift
 		fi
 		;;
-		-l|--label)
-		shift;
-		if [ -n "$1" ]; then
-			label="--label=$1"
-			shift;
+	-l|--label|--labels)
+		shift
+		if [[ -n "${1}" ]]; then
+			labels_arg=" --labels=${1}"
+			shift
 		fi
 		;;
-		-g|--group)
-		shift;
-		if [ -n "$1" ]; then
-			group="--group=$1"
-			shift;
+	-g|--group)
+		shift
+		if [[ -n "${1}" ]]; then
+			group_arg=" --group=${1}"
+			shift
 		fi
 		;;
-		-i|--interval)
-		shift;
-		if [ -n "$1" ]; then
-			default_interval="$1"
-			shift;
+	-i|--interval)
+		shift
+		if [[ -n "${1}" ]]; then
+			default_interval="${1}"
+			shift
 		fi
 		;;
-		-h|--help)
-		usage;
+	--no-install)
+		shift
+		install_arg=" --no-install"
+		;;
+	-h|--help)
+		usage
 		exit 0
 		;;
-		--)
-		shift;
-		break;
+	--)
+		shift
+		break
 		;;
 	esac
 done
 
-typeset -i nerrs=0
-case "$toolset" in
-	default)
-	default_tool_set=$(getconf.py -l default-tool-set pbench/tools)
-	for tool_name in $default_tool_set; do
-		if [[ ${tool_name} == "perf" ]]; then
-			# Ignore, we'll register it after all the others
-			# FIXME: Why is perf always registered?
-			continue
-		fi
-		interval=$(getconf.py interval tools/${tool_name})
-		if [ -z "${interval}" ] ;then
-			interval=${default_interval}
-		fi
-		pbench-register-tool --name=${tool_name} $remote $label $group -- --interval="$interval"
-		rc=$?
-		if [ $rc != 0 ] ;then
-			nerrs=$nerrs+1
+if [[ -z "${toolset}" ]]; then
+	printf -- "ERROR: Missing --toolset=<tool-set> argument\n" >&2
+	usage >&2
+	exit 1
+fi
+
+the_tool_set=$(getconf.py -l ${toolset}-tool-set pbench/tools 2>/dev/null)
+if [[ ${?} -ne 0 ]]; then
+	printf -- "ERROR: failed to fetch tool set, ${toolset}-tool-set, from the pbench-agent configuration file\n" >&2
+	exit 1
+fi
+if [[ -z ${the_tool_set} ]]; then
+	printf -- "ERROR: empty tool set, ${toolset}-tool-set, fetched from the pbench-agent configuration file\n" >&2
+	exit 1
+fi
+
+if [[ -n "${labels_arg}" ]]; then
+	pbench-register-tool --name=perf --test-labels ${remotes_arg}${labels_arg}
```

Comment:
What's `--test-labels`?

Reply:
This is just a way for `pbench-register-tool-set` to check the labels are provided correctly without re-implementing the logic already contained in `pbench-register-tool`.